### DISCUSSION TITLE
Stop Token Consumption

### DISCUSSION THREAD

**Discussion by Rajesh-satan** [vladimir-bukhtoyarov/bucket4j#249] [Q&A]
I have a scenario where i need to maintain order of request for certain exception like "TOO MANY REQUESTS ". Since the requests are sequential if there is an exception after sleep for certain duration order gets mismatched. Is there a way to stop request from consuming tokens and block the subsequent threads.

**Comment by vladimir-bukhtoyarov** [comment]
@Rajesh-satan hello,

If you need to block threads try to play with [Blocking API](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/BlockingBucket.java) - that is accessible via [asBlocking](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/Bucket.java#L45). Describing your question with more details(code snippet is preferable) will be helpful to provide more advices.
